samoa-api/src/main/java/org/apache/samoa/streams/FileStream.java

```diff
@@ -52,9 +49,18 @@
       's', "Source Type (HDFS, local FS)", FileStreamSource.class,
       "LocalFileStreamSource");
 
+  public IntOption classIndexOption = new IntOption("classIndex", 'c',
+          "Class index of data. 0 for none or -1 for last attribute in file.", -1, -1, Integer.MAX_VALUE);
+
+  private FloatOption floatOption = new FloatOption("classWeight", 'w', "", 1.0);
```

Comment:
Yes, it would. The class weight option is indeed implemented via instance weights. 

Most machine learning algorithms focus on total error and in extremely imbalanced scenarios (fraud, terrorism, disease) would fail to detect the sparse class which is really what we're after. Class weighting allows one to incorporate apriori knowledge of the imbalance. For example sklearn, R e1071 SVM packages have class weights options.